sdk/search/azure-search-documents/azure/search/documents/indexes/models/_index.py

```diff
@@ -616,7 +616,7 @@ def _to_generated(self):
         )
 
     @classmethod
-    def _from_generated(cls, search_index):
+    def _from_generated(cls, search_index) -> "SearchIndex":
```

Comment:
Seems like this should be `Optional["SearchIndex"]` since we have the `return None` below?